File: commands/fun/insult.js
const fs = require('fs');
require('random');
const { ApplicationCommandOptionType } = require("discord.js");

module.exports = {
    name: 'insult',
    description: 'Beleidigt jemanden',
    voiceChannel: false,
    blacklist: false,
    options: [
        {
            name: 'user',
            description: 'Wen soll ich beleidigen?',
            type: ApplicationCommandOptionType.User,
            required: false,
        }
    ],

    async execute({ inter }) {

        const fileName = 'insult';
        fs.readFile(fileName, 'utf8', (err, data) => {
            if (err) {
                console.log(err);
                return;
            }
            let insults = data.split('\n')
            let user = inter.options.getUser('user') || inter.user;
            inter.reply(insults[Math.floor(Math.random() * insults.length)].replaceAll("%member%", user));
        });
    },
};

File: Hochverdiener/Commands/Insult.cs
using System.Reflection;
using Discord;
using Discord.WebSocket;
using Hochverdiener_3._0;

namespace Hochverdiener._0.Commands;

public class Insult : BaseCommand
{
    public Insult() : base(
        name: "Insult", 
        aliases: null, 
        description: "Ich beleidige einen User", 
        options: new[] {new SlashCommandOptionBuilder().WithName("user").WithDescription("welchen User ich beleidigen soll").WithType(ApplicationCommandOptionType.User).WithRequired(false)},
        permission: null)
    {
    }
    
    public override Task Execute(SocketSlashCommand command)
    {
        SocketUser user = command.User;
        if(command.Data.Options.Count > 0) user = (SocketUser) command.Data.Options.First().Value;
        command.RespondAsync(ReadInsults()[new Random().Next(0, ReadInsults().Length)].Replace("%member%", user.Mention));
        return base.Execute(command);
    }
    private string[] ReadInsults()
    {
        Assembly assembly = Assembly.GetExecutingAssembly();
        string resourceName = "Hochverdiener_3._0.Content.insults.txt";
        using (Stream? stream = assembly.GetManifestResourceStream(resourceName))
            if (stream != null)
                using (StreamReader reader = new StreamReader(stream))
                {
                    string result = reader.ReadToEnd();
                    string[] insults = result.Split(new[] { Environment.NewLine }, StringSplitOptions.None);
                    return insults;
                }
        return Array.Empty<string>();
    }
}